### PR TITLE
Port remaining tests and remove RSpec

### PR DESCRIPTION
**Pull request by kevindew**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

This converts the few remaining RSpec tests into Minitest and then removes the RSpec installation.

Further details in individual commits.

Closes https://github.com/alphagov/smart-answers/issues/5350

### CODE REVIEW

File: test/functional/flow_controller_test.rb [hunk 1]
```diff
@@ -34,4 +36,174 @@ class FlowControllerTest < ActionController::TestCase
...
+
+      should "remove any answers from the query string" do
+        get :destroy, params: { id: "query-parameters-based", question1: "response1" }
+        assert_redirected_to "/query-parameters-based"
+      end
+    end
+
+    should "redirect to the first question with cache headers for a path based flow" do
+      get :destroy, params: { id: "radio-sample" }
+      assert_redirected_to "/radio-sample/y"
```
Comment (leenagupte) [inline test/functional/flow_controller_test.rb:197]: It's strange that the other flows take you back to the start page, but the path-based flow goes back to the first question.

**Review by leenagupte — COMMENTED** [review]
This is looking good! I've left a couple of inline suggestions, but nothing major.

File: test/functional/flow_controller_test.rb [hunk 2]
```diff
@@ -34,4 +36,174 @@ class FlowControllerTest < ActionController::TestCase
...
+
+  context "GET /<id>/<node_slug>/next" do
+    context "for a session flow" do
+      should "store the answer in the session" do
+        get :update,
+            params: { id: "session-based",
+                      node_slug: "question1",
+                      response: "answer" }
+
+        assert_equal({ "question1" => "answer" }, session["session-based"])
```
[the remark this answers is not in the document]
Comment (kevindew) [reply test/functional/flow_controller_test.rb:100]: Yeah - it's a side-effect that I think should have been captured in https://github.com/alphagov/smart-answers/blob/032c646d119b13d00837b98451ff061201e3660c/spec/requests/session_based_flow_spec.rb#L22-L26

File: test/functional/flow_controller_test.rb [hunk 3]
```diff
@@ -34,4 +36,174 @@ class FlowControllerTest < ActionController::TestCase
...
+
+      assert_redirected_to "/query-parameters-based/question2?question1=response1"
+    end
+
+    should "return a cached response for a query parameter flow" do
+      get :show, params: { id: "query-parameters-based", node_slug: "question1" }
+      assert_cached_response
+    end
+
+    should "return an uncached response for a query parameter flow" do
```
[the remark this answers is not in the document]
Comment (kevindew) [reply test/functional/flow_controller_test.rb]: Oops thanks 👍

File: test/functional/flow_controller_test.rb [hunk 4]
```diff
@@ -34,4 +36,174 @@ class FlowControllerTest < ActionController::TestCase
...
+
+    should "redirect to the first question with cache headers for a path based flow" do
+      get :start, params: { id: "radio-sample" }
+      assert_redirected_to "/radio-sample/y"
+      assert_cached_response
+    end
+  end
+
+  context "GET /<id>/<node_slug>" do
+    should "render a question if the requested question is applicable" do
```
[the remark this answers is not in the document]
Comment (kevindew) [reply test/functional/flow_controller_test.rb]: Thanks, that's better 👍  I did a little tweaking.

File: test/functional/flow_controller_test.rb [hunk 5]
```diff
@@ -34,4 +36,174 @@ class FlowControllerTest < ActionController::TestCase
...
+      get :show, params: { id: "query-parameters-based", node_slug: "question1" }
+      assert_select "h1", "Question 1 title"
+    end
+
+    should "redirect to the current question if a non-applicable question is requested" do
```
[the remark this answers is not in the document]
Comment (kevindew) [reply test/functional/flow_controller_test.rb]: Thanks

File: test/functional/flow_controller_test.rb (hunk 1 above)
Comment (kevindew) [reply test/functional/flow_controller_test.rb:197]: Ooh.. you might not realise it but that just highlighted me to a test description issue: https://github.com/alphagov/smart-answers/compare/ae8906316f9aaea979a8c677d04ba14e1931f187..6f9fae6e2930aebf82bb22efc4584261ed93a62e thanks!

Yeah, this behaviour is different as the path based approach is part of a before_action filter (which, sidenote, I'm never sure of how best to test them - I'm a bit duplication heavy on it in this test) and it kind of makes sense to send them to the first question as they're most likely to experience this when a Smart Answer changes type.

**Comment by kevindew** [comment]
Thanks @leenagupte, all comments should now be addressed.

File: test/functional/flow_controller_test.rb (hunk 1 above)
Comment (leenagupte) [reply test/functional/flow_controller_test.rb:197]: You're welcome?

**Review by leenagupte — APPROVED** [review]
🥇

**Pull request merged**